src/method/method_callback.cpp

```diff
@@ -489,12 +416,30 @@ namespace cubmethod
       }
     else
       {
-	return find_query_handler (iter->second);
+	return get_query_handler_by_id (iter->second);
       }
   }
 
+  query_handler *
+  callback_handler::get_query_handler_by_sql (const std::string &sql)
+  {
+    query_handler *handler = nullptr;
+    for (auto it = m_sql_handler_map.lower_bound (sql); it != m_sql_handler_map.upper_bound (sql); it++)
+      {
+	handler = get_query_handler_by_id (it->second);
+	if (handler != nullptr && handler->get_is_occupied() == false)
+	  {
+	    /* found */
+	    break;
+	  }
+	handler = nullptr;
+      }
+
+    return handler;
```

Comment:
If you do a lot of repetition, I think you can get rid of the substitution statement once.
```
	    /* found */
             return handler;
	  }	
      }

    return nullptr;
```

Reply:
fixed. thank you.